main.go

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2020 Sebastian Pipping <sebastian@pipping.org>
+ * Licensed under AGPL v3 or later
+ */
+package main
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/hartwork/go-wait-for-it/cli"
+	"github.com/hartwork/go-wait-for-it/logging"
+	"github.com/hartwork/go-wait-for-it/syntax"
+)
+
+type ConnectResult struct {
+	address  syntax.Address
+	duration time.Duration
+	success  bool
+}
+
+func waitForAddress(address syntax.Address) chan bool {
+	available := make(chan bool)
+	go func() {
+		for {
+			c, err := net.Dial("tcp", address.String())
+			if err == nil {
```

Comment:
Probably?
```suggestion
			if err != nil {
```

Reply:
And the surface these errors by returning `(bool, error)` with from the functions and not a channel.

Reply:
I re-checked: The condition is not flipped around. I want `waitForAddress` to try connecting to a specific TCP port until it succeeds to no error is the case where I need to close the socket and be done. Right now I don't need the error details and it's an internal function. I think I'll leave it this way.